x-pack/plugin/core/src/main/java/org/elasticsearch/license/StartTrialClusterTask.java

```diff
@@ -58,16 +57,16 @@ public class StartTrialClusterTask implements ClusterStateTaskListener {
 
     private LicensesMetadata execute(
         LicensesMetadata currentLicensesMetadata,
-        DiscoveryNodes discoveryNodes,
+        ClusterState state,
         ClusterStateTaskExecutor.TaskContext<StartTrialClusterTask> taskContext
     ) {
         assert taskContext.getTask() == this;
-        if (discoveryNodes.getMaxNodeVersion().after(discoveryNodes.getSmallestNonClientNodeVersion())) {
+        if (state.nodes().getMaxNodeVersion().after(state.nodes().getSmallestNonClientNodeVersion())) {
```

Comment:
This seems to be using the node version from a different place, but not actually removing its use.

What's the benefit of this check? Why can't a trial license be created in a mixed cluster?

Reply:
I actually did that in response to https://github.com/elastic/elasticsearch/pull/100169#discussion_r1373255542, you have a good point that it's kind of the lazy way of accomplishing this. I probably should have done, uh, a transport version for that?

Reply:
Ahh. I have quite the short memory. :) 

Can we use the new internal feature api? This sounds like exactly the case we had in mind for that api.

Reply:
Oh, derp. Good call on the feature API, that sounds like a good fit. I'll adapt this to use that.